CameraXBasic/app/src/main/java/com/android/example/cameraxbasic/fragments/CameraFragment.kt

```diff
@@ -397,31 +416,30 @@ class CameraFragment : Fragment() {
 
         // Listener for button used to switch cameras
         controls.findViewById<ImageButton>(R.id.camera_switch_button).setOnClickListener {
-            lensFacing = if (CameraX.LensFacing.FRONT == lensFacing) {
-                CameraX.LensFacing.BACK
+            lensFacing = if (CameraSelector.LENS_FACING_FRONT == lensFacing) {
+                CameraSelector.LENS_FACING_BACK
             } else {
-                CameraX.LensFacing.FRONT
-            }
-            try {
-                // Only bind use cases if we can query a camera with this orientation
-                CameraX.getCameraWithLensFacing(lensFacing)
-
-                // Unbind all use cases and bind them again with the new lens facing configuration
-                CameraX.unbindAll()
-                bindCameraUseCases()
-            } catch (exc: Exception) {
-                // Do nothing
+                CameraSelector.LENS_FACING_FRONT
             }
+            // Bind use cases
+            bindCameraUseCases()
         }
 
-        // Listener for button used to view last photo
+        // Listener for button used to view the most recent photo
         controls.findViewById<ImageButton>(R.id.photo_view_button).setOnClickListener {
-            Navigation.findNavController(requireActivity(), R.id.fragment_container).navigate(
-                    CameraFragmentDirections.actionCameraToGallery(outputDirectory.absolutePath))
+            // Only navigate when the gallery has photos
+            if(outputDirectory.listFiles()?.size!! > 0) {
```

Comment:
I would replace this with:
```
if (true == outputDirectory.listFiles()?.isNotEmpty())
```

Reply:
I could reproduce the `IllegalArgumentException` twice in a row (by wildly clicking buttons):

> IllegalArgumentException: navigation destination com.android.example.cameraxbasic:id/action_camera_to_gallery is unknown to this NavController

The other `if` condition might have been better.